go/enclave/l2chain/l2_chain.go

```diff
@@ -929,7 +926,9 @@ func (oc *ObscuroChain) getLatestRollupBeforeBlock(block *common.L1Block) (*core
 
 		block, err = oc.storage.FetchBlock(block.ParentHash())
 		if err != nil {
-			return nil, fmt.Errorf("could not fetch parent block")
+			// No rollup found - no more blocks available (enclave does not read the L1 chain from genesis if it knows
+			// when management contract was deployed, so we don't keep going to block zero, we just stop when the blocks run out)
+			return nil, errNoRollupFound
```

Comment:
Wait isn't this the block that was not found ?

Reply:
It was scanning backwards through blocks looking for a rollup. If it runs out of blocks then there was no rollup. I'll update the comment to try to make that clearer

Reply:
Also should we surface that block error ?